azurerm/helpers/azure/hdinsight.go

```diff
@@ -156,6 +156,34 @@ func SchemaHDInsightsExternalMetastore() *schema.Schema {
 	}
 }
 
+func SchemaHDInsightsMonitor() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"log_analytics_workspace_id": {
+					Type:     schema.TypeString,
+					Required: true,
+					ForceNew: false,
+				},
```

Comment:
HDInisght needs `workspace_id` (https://www.terraform.io/docs/providers/azurerm/r/log_analytics_workspace.html#workspace_id) property that just guid instead of `id` that is proper Azure Resource ID that can be validated with `LogAnalyticsWorkspaceID` func